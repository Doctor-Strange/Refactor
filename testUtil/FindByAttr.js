

export const FindByAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`);
};
