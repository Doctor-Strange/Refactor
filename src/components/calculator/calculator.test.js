import React from "react";
import { mount } from "enzyme";
import { FindByAttr } from "../../../testUtil/FindByAttr";
import Calculator from "./";

const setup = (props) => {
  return mount(<Calculator {...props} />);
};

describe("Calculator component", () => {
  describe("snapshot", () => {
    test("/components/calculator", () => {
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });

//   describe("AB test props", () => {
//     test("check value of anchor tag", () => {
//       const wrapper = setup({
//         AbText: "از درخواست‌های اجاره مرتبط باخبر شوید",
//       });
//       const form = FindByAttr(wrapper, "form");
//       form.simulate("submit")
//       const addCar_top_joinus_a = FindByAttr(wrapper, "addCar_top_joinus_a");
//       expect(addCar_top_joinus_a.text()).toEqual(
//         "از درخواست‌های اجاره مرتبط باخبر شوید"
//       );
//     });
//   });
});
