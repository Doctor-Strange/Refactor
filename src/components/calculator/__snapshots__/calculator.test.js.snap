// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component snapshot /components/calculator 1`] = `ReactWrapper {}`;
